src/test/java/org/assertj/core/api/inputstream/InputStreamAssert_asString_with_charset_Test.java

```diff
@@ -47,6 +54,23 @@ void should_convert_inputStream_to_a_proper_string_with_specific_encoding() {
                      .isEqualTo(real);
   }
 
+  @Test
+  void should_throw_exception() throws IOException {
+    // GIVEN
+    String real = "Gerçek";
+    InputStream inputStream = mock(InputStream.class);
+
+    // MOCK BEHAVIOUR
+    when(inputStream.read(any(), anyInt(), anyInt())).thenThrow(new IOException());
+
+    // ASSERT THROWS
+    assertThatThrownBy(() -> {
+      // WHEN/THEN
+      then(inputStream).asString(TURKISH_CHARSET)
+        .isEqualTo(real);
+    }).isInstanceOf(InputStreamsException.class);
```

Comment:
Minor: restructure the test from the ASSERT THROWS comment as:
```java
   // WHEN/THEN
   thenThrownBy(() -> assertThat(inputStream).asString(TURKISH_CHARSET).isEqualTo(real)).isInstanceOf(InputStreamsException.class);
``` 
an alternative would be:
```java
   // WHEN
   ThrowingCallable assertionCall = () -> assertThat(inputStream).asString(TURKISH_CHARSET).isEqualTo(real);
   // WHEN/THEN
   thenThrownBy(assertionCall).isInstanceOf(InputStreamsException.class);
```
pick the one that is most readable to you ;-)